pkg/pf/tests/internal/cross-tests/tfwrite.go

```diff
@@ -1,173 +1,182 @@
+	pfNestingModeSingle  pfNestingMode = 3
+)
+
+func pfNestingToShim(nesting pfNestingMode) crosstestsimpl.Nesting {
+	switch nesting {
+	case pfNestingModeSingle:
+		return crosstestsimpl.NestingSingle
+	case pfNestingModeList:
+		return crosstestsimpl.NestingList
+	case pfNestingModeSet:
+		return crosstestsimpl.NestingSet
+	default:
+		return crosstestsimpl.NestingInvalid
 	}
-	f := hclwrite.NewEmptyFile()
-	block := f.Body().AppendNewBlock("resource", []string{resourceType, resourceName})
-	writePfResource(block.Body(), sch, config)
-	_, err := f.WriteTo(out)
-	return err
-}
-
-func writePfProvider(body *hclwrite.Body, schemas pschema.Schema, values map[string]cty.Value) {
-	writePfObjectProvider(body, pschema.NestedBlockObject{
-		Attributes: schemas.Attributes,
-		Blocks:     schemas.Blocks,
-	}, values)
-}
-
-func writePfResource(body *hclwrite.Body, schemas rschema.Schema, values map[string]cty.Value) {
-	writePfObjectResource(body, rschema.NestedBlockObject{
-		Attributes: schemas.Attributes,
-		Blocks:     schemas.Blocks,
-	}, values)
-}
-
-// writePfBlockProvider writes the values for a single schema block to parentBody.
-//
-// Because blocks can be repeated (ListNestedBlock and SetNestedBlock), writePfBlockProvider
-// can write an arbitrary number of blocks.
-//
-// For example, writing a list would add two blocks to parentBody:
-//
-//	writePfBlockProvider("key", parentBody, ListNestedBlock{count: int}, cty.Value([{count: 1}, {count: 2}]))
-//
-//	key {
-//	  count = 1
-//	}
-//	key {
-//	  count = 2
-//	}
-//
-// This is why writePfBlockProvider is called with parentBody, instead of with the block body
-// already created (as with [writeBlock]).
-func writePfBlockProvider(key string, parentBody *hclwrite.Body, schemas pschema.Block, value cty.Value) {
-	switch schemas := schemas.(type) {
+}
+
+func pSchemaBlockToObject(block pschema.Block) pschema.NestedBlockObject {
+	switch block := block.(type) {
 	case pschema.ListNestedBlock:
-		for _, v := range value.AsValueSlice() {
-			b := parentBody.AppendNewBlock(key, nil).Body()
-			writePfObjectProvider(b, schemas.NestedObject, v.AsValueMap())
-		}
+		return block.NestedObject
 	case pschema.SetNestedBlock:
-		values := value.AsValueSet().Values()
-		for _, v := range values {
-			b := parentBody.AppendNewBlock(key, nil).Body()
-			writePfObjectProvider(b, schemas.NestedObject, v.AsValueMap())
-		}
+		return block.NestedObject
 	case pschema.SingleNestedBlock:
-		body := parentBody.AppendNewBlock(key, nil).Body()
-
-		if value.IsNull() {
-			return
+		return pschema.NestedBlockObject{
+			Attributes: block.Attributes,
+			Blocks:     block.Blocks,
 		}
-
-		writePfObjectProvider(body, pschema.NestedBlockObject{
-			Attributes: schemas.Attributes,
-			Blocks:     schemas.Blocks,
-		}, value.AsValueMap())
 	default:
-		contract.Failf("Unknown block type: %T", schemas)
+		contract.Failf("Unknown block type: %T", block)
 	}
+	return pschema.NestedBlockObject{}
```

Comment:
Nit: lets move the `return` into the default block.
```suggestion
		contract.Failf("Unknown block type: %T", block)
		return pschema.NestedBlockObject{}
	}
```